pkg/fetcher/resource.go

```diff
@@ -243,6 +247,55 @@ func Other(cause error) *ResourceError {
 	}
 }
 
+func HTTPStatusToException(status int) *ResourceError {
+	if status == 0 {
+		return nil
+	}
+
+	switch status {
+	case http.StatusOK, http.StatusCreated, http.StatusAccepted, http.StatusPartialContent, http.StatusNoContent, http.StatusResetContent, http.StatusNotModified:
+		return nil
+	default:
+		return NewResourceError(ResourceErrorCode(status))
+	}
+}
+
+func AWSErrorToException(err error) *ResourceError {
+	var notFound *types.NotFound
+	var noSuchKey *types.NoSuchKey
+	var noSuchBucket *types.NoSuchBucket
+	var invalidObjectState *types.InvalidObjectState
+	if errors.As(err, &notFound) || errors.As(err, &noSuchKey) || errors.As(err, &noSuchBucket) {
+		return NotFound(err)
+	} else if errors.As(err, &invalidObjectState) {
+		return BadRequest(err)
+	} else {
+		var ae smithy.APIError
+		if errors.As(err, &ae) {
+			if ae.ErrorCode() == "AccessDenied" {
+				return Forbidden(err)
+			}
+		}
+	}
+
+	return Other(err)
+}
+
+func GCSErrorToException(err error) *ResourceError {
```

Comment:
These don't look like they belong in this module?

Reply:
I put them here because it's the best place I could think of, since it's not like we have a special package for remote/cloud streaming. I was thinking about creating `pkg/cloud`, but it seemed to bare-bones. What do you think would be the best place to place these?

Reply:
I would put `AWSErrorToException` in `fetcher_s3.go` and `GCSErrorToException` in `fetcher_gcs.go` where they are used, and probably keep them private.